### PR TITLE
Removed Matrix, Vector module and refactored

### LINKED ISSUE #4: Move private function `and_then2` to own module.
Function `and_then2` is defined in two module `lib/qr_code/matrix.ex` and `lib/qr_code/pattern.ex`. This is antipattern.

### PR DESCRIPTION
**Pull request by ondrej-tucek**
Resolve #4 and #6

### CODE REVIEW

**Comment by ondrej-tucek** [comment]
It's interesting that dialyzer didn't catch this at `MatrixReloaded` package...

File: lib/qr_code/masking.ex [hunk 1]
```diff
@@ -35,6 +35,134 @@ defmodule QRCode.Masking do
...
+  defp compute_penalty_2(rows, acc \\ 0)
+
+  defp compute_penalty_2(rows, acc) when length(rows) == 1 do
+    acc
+  end
+
+  defp compute_penalty_2([row1, row2 | rows], acc) do
+    acc_row =
+      row1
+      |> map2(row2, fn v1, v2 -> v1 + v2 end)
```
Comment (s-m-i-t-a) [inline lib/qr_code/masking.ex]: Why use new `map2` function instead `Enum.zip/2` and lambda function `fn {v1, v2} -> v1 + v2 end` in `Enum.map`?

File: mix.exs [hunk 2]
```diff
@@ -6,7 +6,7 @@ defmodule QRCode.MixProject do
       app: :qr_code,
       dialyzer: dialyzer_base() |> dialyzer_ptl(System.get_env("SEMAPHORE_CACHE_DIR")),
       version: "0.1.0",
-      elixir: "~> 1.7",
+      elixir: "~> 1.8",
```
Comment (s-m-i-t-a) [inline mix.exs]: We use some incompatibity from *1.8*?

File: lib/qr_code/masking.ex [hunk 3]
```diff
@@ -35,6 +35,134 @@ defmodule QRCode.Masking do
...
+      Kernel.abs(percent_of_dark - reminder - 45) / 5
+    ) * 10
+  end
+
+  defp compute_penalty_1(matrix) do
+    matrix
+    |> Enum.reduce(0, fn [h | _] = row, acc ->
+      row
+      |> Enum.reduce({h, 0, acc}, &evaluate_cond_1/2)
+      |> (fn {__selected, _sum, val} -> val end).()
```
Comment (s-m-i-t-a) [inline lib/qr_code/masking.ex]: Pls, use `Kernel.elem/2` function instead, looks better.

File: mix.exs (hunk 2 above)
Comment (ondrej-tucek) [reply mix.exs]: At lower elixir version (<1.8) is not functional `floor` function in guards.

File: lib/qr_code/masking.ex (hunk 3 above)
Comment (ondrej-tucek) [reply lib/qr_code/masking.ex]: Oki

File: lib/qr_code/masking.ex (hunk 1 above)
Comment (ondrej-tucek) [reply lib/qr_code/masking.ex]: Hmm I really don't know now :-) why I used `map2` approach instead of `Enum.zip`...

**Comment by ondrej-tucek** [comment]
Module `QRCode.Utils` can be deleted, but before that put the funciton `put_to_list(el, tpl)` as private to the `Placement` module and save function can be added to readme...

**Pull request merged**